### ISSUE TITLE
Not calling next function on zoomed out screens

### ISSUE THREAD

**Issue by finishedcoot** [ankeetmaini/react-infinite-scroll-component#380]
When the window is zoomed out and the component is initialized the component will stuck on loading wont call the next function

**Comment by techwithanirudh** [comment]
@finishedcoot I'm having the same issue? What's the fix?

**Comment by techwithanirudh** [comment]
@finishedcoot How could I use your fork without having yarn installed?

**Comment by finishedcoot** [comment]
@techwithanirudh @tofiqquadri Temporarily you can install the package by changing your `package.json` to be like this and reinstall the module.
But I recommend to change it back after the maintainer fixed this issue ASAP.

```package.json
"react-infinite-scroll-component": "git+https://github.com/Stacrypt/react-infinite-scroll-component",
```

**Comment by LibardoVega** [comment]
hola yo pude resolver el hecho que no llame a la función "next", lo que me daba error era usar el componente dentro de una etiqueta que usara "Overflow: auto"

**Comment by LibardoVega** [comment]
acabe de revisar bien la documentación y hace falta el "scrollableTarget" con eso ya pude usar el "Overflow: auto"

**Comment by tofiqquadri** [comment]
I avoided using this library since it has this major bug open and it has not yet solved.

Instead of this I used IntersectionObserver and implemented the functionality of this library myself.

**Comment by kimkong88** [comment]
I am too waiting on this!

**Comment by Fernando-Hooklab** [comment]
I'm having the same issue!

**Comment by kimkong88** [comment]
> I'm having the same issue!

my workaround was just to fetch enough data for single page OR have each item card large enough to fit scroll.. OR load more manually(not waiting for scroll to reach the bottom) based on height

**Comment by ryanschiang** [comment]
Issue still persists. See this codepen for example: [https://codesandbox.io/s/amazing-hermann-x2qw8m?file=/src/index.js](https://codesandbox.io/s/amazing-hermann-x2qw8m?file=/src/index.js)

Unfortunately this cripples the library for anything besides simple/single-page infinite scroll apps.

**Comment by ryanschiang** [comment]
Here's my workaround InfiniteScroll component, using IntersectionObserver:

`interface InfiniteScrollProps {
  load: () => void;
  hasMore: boolean;
  loader: React.ReactNode;
  children?: React.ReactNode;
  endMessage?: React.ReactNode;
}

export const InfiniteScroll: React.FC<InfiniteScrollProps> = ({
  load,
  hasMore,
  loader,
  children,
  endMessage,
}) => {
  const sentinelRef = useRef<HTMLDivElement>(null);
  const observerRef = useRef<IntersectionObserver | null>(null);

  const handleIntersect = useCallback(
    (
      entries: IntersectionObserverEntry[],
      observer: IntersectionObserver
    ) => {
      // Check if the sentinel element is intersecting, and if so, call the load function
      if (entries[0].isIntersecting && hasMore) {
        load();
      }
    },
    [load]
  );

  useEffect(() => {
    // Create a new IntersectionObserver when the component mounts
    observerRef.current = new IntersectionObserver(handleIntersect, {
      root: null,
      rootMargin: "0px",
      threshold: 1.0,
    });

    // Attach the observer to the sentinel element
    if (sentinelRef.current) {
      observerRef.current.observe(sentinelRef.current);
    }

    // Clean up the observer when the component unmounts
    return () => {
      if (observerRef.current) {
        observerRef.current.disconnect();
      }
    };
  }, [load]);

  useEffect(() => {
    // When the hasMore prop changes, disconnect the previous observer and reattach it to the new sentinel element
    if (observerRef.current && sentinelRef.current) {
      observerRef.current.disconnect();
      observerRef.current.observe(sentinelRef.current);
    }
  }, [hasMore]);

  return (
    <div>
      {children}
      <div ref={sentinelRef}>{hasMore && loader}</div>
      {!hasMore && endMessage}
    </div>
  );
};`

**Comment by eamador** [comment]
You can workaround this issue with a hook, check the following response

https://github.com/ankeetmaini/react-infinite-scroll-component/issues/391#issuecomment-1734154621

**Comment by fridaystreet** [comment]
> IntersectionObserver

any chance you could expand on that a bit?

**Comment by Zemelia** [comment]
I did the next workaround, and it works fine so far...

```jsx
const myRef = useRef<HTMLDivElement>(null);

useLayoutEffect(() => {
  const element = myRef.current;
  if (
    hasMore &&
    element?.firstChild &&
    element?.clientHeight >= element?.scrollHeight
  ) {
    const resizeObserver = new ResizeObserver((entries) => {
      if (entries[0].target.clientHeight <= element.clientHeight) {
        element.dispatchEvent(new CustomEvent('scroll'));
      }
    });

    resizeObserver.observe(element.firstChild as Element);
    return () => resizeObserver.disconnect();
  }
}, [hasMore]);

return (
  <div id="scroller-wrapper" ref={myRef}>
    <InfiniteScroll
      dataLength={dataLength}
      next={moreData}
      hasMore={hasMore}
      scrollThreshold="50px"
      scrollableTarget="scroller-wrapper"
    >
      {children}
    </InfiniteScroll>
  <div>
)

```

**Comment by iamdarshshah** [comment]
This is now fixed in v7.1.0. As several folks in this thread independently discovered, `IntersectionObserver` is the right fix, and that is exactly what v7.1.0 ships. A sentinel element is placed at the bottom of the list and observed. When it enters the viewport (including on zoom-out or long screens where no scroll exists), `next()` fires automatically. All the custom `IntersectionObserver` workarounds shared here are no longer needed. Upgrading to v7.1.0 will resolve this. Feel free to reopen if you are still facing this after upgrading.
